store/tikv/2pc.go

```diff
@@ -712,11 +712,23 @@ func (actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 			}
 			locks = append(locks, lock)
 		}
-		_, err = c.store.lockResolver.ResolveLocks(bo, c.startTS, locks)
+		var expire int64
+		expire, err = c.store.lockResolver.ResolveLocks(bo, c.startTS, locks)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		// Because we already waited on tikv, no need to Backoff here.
+
+		if err1 := bo.BackoffWithMaxSleep(boTxnLockFast, int(expire), errors.New(locks[0].String())); err1 != nil {
+			return err1
+		}
+		// Handle the killed flag when waiting for the pessimistic lock.
+		// When a txn runs into LockKeys() and backoff here, it has no chance to call
+		// executor.Next() and check the killed flag.
+		if action.killed != nil {
+			if atomic.CompareAndSwapUint32(action.killed, 1, 0) {
```

Comment:
It is possible that handleSingleBatch error is shadowed by another single batch error, then this error may be ignored.
Can we just Load rather than CAS?

Reply:
`LockKeys()` is a synchronous operation @coocood

Reply:
This is not `LockKeys` function, it's run on its own goroutine.

Reply:
Done